v2/devdoc/log_context_property_basic_types_requirements.md

```diff
@@ -29,392 +29,585 @@ Note: More types will be added as needed.
     uint8_t
 
 #define DECLARE_BASIC_TYPE(type_name) \
-    extern const LOG_CONTEXT_PROPERTY_TYPE_IF MU_C2(type_name, _log_context_property_type);
+    int LOG_CONTEXT_PROPERTY_TYPE_INIT(type_name)(void* dst_value, type_name src_value); \
+    size_t LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(type_name)(type_name src_value); \
+    extern const LOG_CONTEXT_PROPERTY_TYPE_IF LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(type_name); \
 
 MU_FOR_EACH_1(DECLARE_BASIC_TYPE, SUPPORTED_BASIC_TYPES)
 ```
 
-## LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).to_string
+## LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).to_string
 
 ```c
 typedef int (*LOG_CONTEXT_PROPERTY_TYPE_TO_STRING)(void* property_value, char* buffer, size_t buffer_length);
 ```
 
-`LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).to_string` produces the string representation of an `int64_t`.
+`LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).to_string` produces the string representation of an `int64_t`.
 
-**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_001: [** If `property_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).to_string` shall fail and return a negative value. **]**
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_001: [** If `property_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).to_string` shall fail and return a negative value. **]**
 
-**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_002: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).to_string` shall call `snprintf` with `buffer`, `buffer_length` and format string `PRId64` and pass in the values list the `int64_t` value pointed to be `property_value`. **]**
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_002: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).to_string` shall call `snprintf` with `buffer`, `buffer_length` and format string `PRId64` and pass in the values list the `int64_t` value pointed to be `property_value`. **]**
 
-**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_003: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).to_string` shall succeed and return the result of `snprintf`. **]**
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_003: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).to_string` shall succeed and return the result of `snprintf`. **]**
 
-**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_004: [** If any error is encountered, `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).to_string` shall fail and return a negative value. **]**
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_004: [** If any error is encountered, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).to_string` shall fail and return a negative value. **]**
 
-## LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).copy
+## LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).copy
 
 ```c
 typedef int (*LOG_CONTEXT_PROPERTY_TYPE_COPY)(void* dst_value, void* src_value);
```

Comment:
>void [](http://example.com/codeflow?start=63&length=4)

should be const, as we're not taking ownership of it.

Reply:
Fixed.